changelog.d/21707_exclusive_route_transform.feature.md

```diff
@@ -0,0 +1,3 @@
+Introduce a new exclusive_route transform, which functions as a switch statement to route events based on user-defined conditions.
```

Comment:
I think you mentioned you'd be writing a blog post on this one. I think a release highlight might be appropriate too (or instead of the blog): https://vector.dev/highlights/.